keps/sig-multicluster/1645-multi-cluster-services-api/specification.md

```diff
@@ -0,0 +1,263 @@
+
+
+### 2.2 - Record for Schema Version
+
+Following the existing specification, clusters implementing multicluster DNS will contain an additional `TXT` record responding with the semantic version of the DNS schema used for the multi cluster DNS `<zone>`, also known in this specification as `<clustersetzone>`.
+
+- Question Example:
+  - `dns-version.clusterset.local. IN TXT`
+- Answer Example:
+  - `dns-version.clusterset.local. 28800 IN TXT “1.0.0”`
+
+### 2.3 - Records for a Service with ClusterSetIP
+
+#### 2.3.1 - `A`/`AAAA` Record
+
+_Note: This section refers to `A` and `AAAA` record requirements. For Service objects that have dual-stack networking enabled, both `A` and `AAAA` records must be created to cover both IPv4 and IPv6 assigned ClusterSetIPs._
+
+Given a ClusterIP type Service named `<service>` in Namespace `<ns>` that has been exported via a name-mapped ServiceExport with name `<service>`, given it is accessible across the clusterset by the IP address `<clusterset-ip>`, the following records must exist.
+
+If the `<clusterset-ip>` is an IPv4 address, an `A` record of the following form must exist.
+
+
+
+*   Record Format:
+    *   `<service>.<ns>.svc.<clustersetzone>. <ttl> IN A <clusterset-ip>`
+*   Question Example
+    *   `myservice.test.svc.clusterset.local. IN A`
+*   Answer Example:
+    *   `myservice.test.svc.clusterset.local. 4 IN A 10.42.42.42`
+
+If the `<clusterset-ip>` is an IPv6 address, an `AAAA` record of the following form must exist.
+
+
+
+*   Record Format:
+    *   `<service>.<ns>.svc.<clustersetzone>. <ttl> IN AAAA <clusterset-ip>`
+*   Question Example:
+    *   `myservice.test.svc.clusterset.local. IN AAAA`
+*   Answer Example:
+    *   `myservice.test.svc.clusterset.local. 4 IN AAAA 2001:db8::1`
+
+
+#### 2.3.2 - `SRV` Records
+
+For each port in an exported Service with name `<port>` and number `<port-number>` using protocol `<proto>`, an `SRV` record of the following form must exist.
+
+
+
+*   Record Format:
+    *   `_<port>._<proto>.<service>.<ns>.svc.<clustersetzone>. <ttl> IN SRV <weight> <priority> <port-number> <service>.<ns>.svc.<zone>.`
+
+The priority `<priority>` and weight `<weight>` are numbers as described in [RFC2782](https://tools.ietf.org/html/rfc2782) and whose values are not prescribed by this specification.
+
+Unnamed ports do not have an `SRV` record.
+
+
+
+*   Question Example:
+    *   `_https._tcp.myservice.test.svc.clusterset.local. IN SRV`
+*   Answer Example:
+    *   `_https._tcp.myservice.test.svc.clusterset.local. 30 IN SRV 10 100 443 myservice.test.svc.clusterset.local.`
+
+The Additional section of the response may include the Service `A`/`AAAA` record referred to in the `SRV` record.
+
+#### 2.3.3 - `PTR` Record
```

Comment:
Maybe this was discussed before - do we know if there is value in PTR records for ClusterIP services? cc @thockin @johnbelamaric 
It makes sense for headless services, but I am curious if we need this for non-headless, both in multi-cluster and cluster-local cases.

Reply:
Thanks for the video reference of this discussion! I am leaning towards the "not required" approach since that is the least invasive. Rewriting existing cluster.local PTR records with clusterset.local could be problematic in case someone was relying on those, especially if rewrites will happen for headless service + pod PTR records. 

Given that 1) cluster.local PTR records will exist 2) user can implement a dummy service in importing clusters 3) user can invoke k8s api to get the same or more info, not requiring additional PTR records for clusterset.local sounds reasonable to me.

Reply:
Fixed in e7dd34d by not requiring PTR records for multicluster DNS. I added in a lot of my comment above as rationale in the MCS API KEP so it is easy to find/immortalized.